### PR TITLE
Upgrade Bioregistry import

### PR DESCRIPTION
**Pull request by cthoyt**
This PR updates the Bioregistry import to handle Bioregistry v0.6.0 and above. This new version makes it possible to specify a priority order for prefixes. It also implements better functionality for switching between modern and legacy Identifiers.org URIs.

This PR does not yet take advantage of prefix synonyms, since that would make the diff much larger and more complicated.

### CODE REVIEW

File: src/prefixmaps/ingest/ingest_bioregistry.py [hunk 1]
```diff
@@ -54,24 +46,29 @@ def from_bioregistry(upper=False, canonical_idorg=True, filter_dubious=True) ->
...
+    context = Context("bioregistry", upper=upper)
+    prefix_priority = [
+        #  "obofoundry.preferred",
+        "preferred",
+        # "obofoundry",
+        "default",
+    ]
+    priority = [
+        "obofoundry",
+        "miriam.legacy" if canonical_idorg else "miriam",
```
Comment (sierra-moxon) [inline src/prefixmaps/ingest/ingest_bioregistry.py:58]: do we need semweb here?

**Comment by sierra-moxon** [comment]
Hi Charlie - I also thought about adding "gro/GRO" to a skip list, but I didn't want to hard-code something here.  Is the "deprecated" flag meant differently in bioregistry?  It seems like we don't want to promote the use of a deprecated prefix?

File: src/prefixmaps/ingest/ingest_bioregistry.py (hunk 1 above)
Comment (sierra-moxon) [reply src/prefixmaps/ingest/ingest_bioregistry.py:62]: bioregistry contains these right?  Is the full list below in the registry?  (I use this context file in Biolink, if I want to replace its import with the "merged" file from this repo, I need to make sure we still have all the prefixes, just coming from a different source).  I know you have some graphics that show the overlap between prefixcommons and bioregistry -- can you point me to that doc or a list of outstanding prefixes in prefixcommons that aren't in the registry if any?)
 https://github.com/prefixcommons/biocontext/blob/master/registry/semweb_context.yaml

**Comment by cthoyt** [comment]
@sierra-moxon I think maintaining a short manual skip list is a better stop-gap than excluding all deprecated resources. It's surprising how many deprecated resources are still in wide use.

I'll take another look after this on why this prefix is having so many problems and hopefully eliminate the need for a skip list after merge of this PR.

File: src/prefixmaps/ingest/ingest_bioregistry.py (hunk 1 above)
Comment (cthoyt) [reply src/prefixmaps/ingest/ingest_bioregistry.py:62]: - List of metaregistries: https://bioregistry.io/metaregistry
- Overlap summary: https://bioregistry.io/summary

The `bioregistry.get_extended_prefix_map` priority keywords take valid metaprefixes from the metaregistries and a small number of special cases like `default`, `miriam.legacy`, and `preferred` that signal some special logic during mapping. 

You might be mixing this up with Bioregistry's collections (https://bioregistry.io/collection/), which just allows for task-specific or topic-specific lists of canonical Bioregistry prefixes to be grouped together. This does not factor into anything here.

Comment (cthoyt) [reply src/prefixmaps/ingest/ingest_bioregistry.py:62]: But if the question was "does the bioregistry contain the prefixes that are in this other semantic web list" then the answer is yes. You can see the semantic web collection to check https://bioregistry.io/collection/0000002. Let me know if something's missing.

File: src/prefixmaps/data/bioregistry.csv [hunk 2]
```diff
@@ -126,23 +120,16 @@ bioregistry,BSPO,http://purl.obolibrary.org/obo/BSPO_,canonical
...
-bioregistry,cabri,http://identifiers.org/cabri/,canonical
-bioregistry,cadsr,http://identifiers.org/cadsr/,canonical
 bioregistry,caloha,https://biopragmatics.github.io/providers/caloha/,canonical
 bioregistry,cameo,http://identifiers.org/cameo/,canonical
 bioregistry,caps,http://identifiers.org/caps/,canonical
 bioregistry,CARO,http://purl.obolibrary.org/obo/CARO_,canonical
 bioregistry,cas,http://identifiers.org/cas/,canonical
 bioregistry,cath,http://identifiers.org/cath/,canonical
 bioregistry,cath.domain,http://identifiers.org/cath.domain/,canonical
-bioregistry,cath.superfamily,http://identifiers.org/cath.superfamily/,canonical
```
Comment (cmungall) [inline src/prefixmaps/data/bioregistry.csv:140]: is there a reason this is disappearing?

Comment (cthoyt) [reply src/prefixmaps/data/bioregistry.csv:140]: Its URI prefix is redundant of `cath` (though this is obscured by using identifiers.org URIs)

Comment (cmungall) [reply src/prefixmaps/data/bioregistry.csv:140]: I see. I find the subspacing decisions (inherited from identifiers.org) confusing and inconsistent. Sometimes the top level corresponds to a subspace, sometimes it is its own thing. But we can take this to the existing ticket: https://github.com/biopragmatics/bioregistry/issues/133

I don't love prefixes or URLs disappearing, we should have guards against this in future but I think some churn in the initial phase is acceptable.

**Review by sierra-moxon — APPROVED** [review]

**Pull request merged**